blog/content/post/2020-03-27-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Open Source DRTM with TrenchBoot. Landing Zone validation.
+abstract: When you already know what is TrenchBoot, what is DRTM and how we
+          enable it on AMD processors, we can move on to practice. Today, I will
+          show you how to configure all components and verify first project's
+          requirements.
+cover: /covers/
+author: piotr.kleinschmidt
+layout: post
+published: false
+date: 2020-03-27
+archives: "2020"
+
+tags:
+  - trenchboot
+  - security
+  - open-source
+  - coreboot
+categories:
+  - Firmware
+  - Security
+
+---
+
+In [previous article](link-to-tb-nlnet-basics) I introduced project's basics. I
+explained briefly what parts of system are necessary in DRTM and how to prepare
+them. Now, let's try to build them, so you can enjoy having secure platform too.
+Also, we will verify first requirements which are already met in project.
+
+## Preparation
+
+We are using PC Engines apu2 platform with coreboot and NixOS. Procedures which
+are presented further in article are done for that exact configuration. If you
+want to perform any operations step-by-step, I strongly recommend to use exactly
+the same setup. I assume you have already installed NixOS and can install
+original and custom packages (tutorials are in [previous article](link-to-tb-nlnet-basics)).
+It's important because it will be used very often, so you shouldn't have any
+problems with it.
+
+Also in NixOS download our [custom `nixpkgs` library](linkt-to-custom-nixpkgs).
+It already has every additional tool and component added, so it will be easy to
+install it.
+
+
+## Verification
+
+To create platform with DRTM enabled and properly prepared Linux kernel you must
+install following packages:
+
+- linux-kernel-trenchboot
+- initrd-trenchboot
+- GRUB-trenchboot
+- LZ-trenchboot
+
+Reboot the platform and see if NixOS boots normally. Later in the article, I
+will show you how to verify each part of above operation and hence DRTM itself.
+
+### Linux kernel and initrd
+
+Prepare system by installing following packages: `linux-kernel-trenchboot` and
+`initrd-trenchboot`. Rebuild kernel and reboot platform.
+
+```
+What to exactly verify?
+What should be/be not seen in NixOS?
+```
+
+### GRUB
+
+There are two ways to validate if GRUB will load `slaunch` module and hence run
+SKINIT and LZ (DRTM). First, during platform boot you can see if there is
+`slaunch` option available and if you can boot from it. Second, you can verify
+content of `grub.cfg` file. Both methods with possible outputs are presented
+below.
+
+#### GRUB without slaunch
+
+Prepare your system. Make sure that `GRUB-trenchboot` package is not
```

Comment:
@piotr-kleins we can still leave the default NixOS kernel and old menu entry from GRUb to boot old kernel. Nothing has to be uninstalled

Reply:
Outdated